### ISSUE TITLE
no-redeclare: 1.79.0 regression — flags legal TypeScript value+type declaration merging (`export const X` + `export type X`)

### ISSUE THREAD

**Issue by ahmedelgabri** [oxc-project/oxc#25936]
Upgrading 1.78.0 → 1.79.0 introduced a false positive in `no-redeclare` on TypeScript files. This six-line file is enough to reproduce (`no-redeclare: error`):

```ts
export const Level = {
	Info: 'info',
	Error: 'error',
} as const

export type Level = (typeof Level)[keyof typeof Level]
```

On 1.79.0:

> `case.ts:1:14: error eslint(no-redeclare): 'Level' is already defined.`

On 1.78.0 the identical file exits 0.

The pattern is valid TypeScript — a value and a type sharing a name is declaration merging, and this const-object-as-enum shape is one of the most common idioms built on it. typescript-eslint's `@typescript-eslint/no-redeclare` accepts it for the same reason (`ignoreDeclarationMerge`, on by default).

Expected: on TypeScript sources, `no-redeclare` ignores value+type merging, as it did through 1.78.0.

**Comment by camc314** [comment]
https://typescript-eslint.io/play/#ts=6.0.2&showAST=scope&fileType=.tsx&code=FAUwHgDg9gTgLgAgMZQHYGdEBkQDcQA2CAvAgN7AIICSqAZlAFwIDkAlvVCwDSUICiMGLGYsQQ2D2ABfBAEN0yNJmChIsRHACeEEAhz4ipABTbdUOvryEAlAG0A1iC0WEZkK4OEAukA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1ukQBNEy8AIY90URNGgdokcGAC%2BIeUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

This behaviour aligns with the upstream typescript-eslint rule.

I would recommend disabling this rule for typescript files as it is checked already by the typescript compiler.
